repos/system_upgrade/common/models/repositoriesmap.py

```diff
@@ -2,34 +2,93 @@
 from leapp.topics import TransactionTopic
 
 
-class RepositoryMap(Model):
+class RepoMapEntry(Model):
+    topic = TransactionTopic
+
+    source = fields.String()
+    """The source PES id."""
+
+    target = fields.List(fields.String())
+    """List of target PES ids"""
+
+
+class PESIDRepositoryEntry(Model):
     """
-    Mapping between repositories to be used during upgrade.
+    Represent metadata about particular repository.
 
-    Mapping of current system repository to target system version repositories to determine which
-    ones should be enabled for the correct upgrade process.
+    The metadata are used to identify purpose and nature of the repository.
+
+    Warning: This model is not expected to be consumed/produced by any actor
+    directly. As well, it is not covered by deprecation process and can be
+    changed or removed any time.
     """
 
     topic = TransactionTopic
 
-    from_repoid = fields.String()
-    """Source RHEL repoid as present in the Red Hat CDN"""
-    to_repoid = fields.String()
-    """Target RHEL repoid as present in the Red Hat CDN"""
-    to_pes_repo = fields.String()
-    """Target RHEL repo name as used in the Package Evolution Service database"""
-    from_minor_version = fields.String()
-    """To which source RHEL minor versions the mapping relates to"""
-    to_minor_version = fields.String()
-    """To which target RHEL minor versions the mapping relates to"""
-    arch = fields.String()
-    """CPU architecture the mapping relates to"""
-    repo_type = fields.StringEnum(choices=['rpm', 'srpm', 'debuginfo'])
-    """Type of content the mapped repos hold"""
-
-
-class RepositoriesMap(Model):
+    pesid = fields.String()
+    """
+    The PES id of the repository.
+
+    The PES id indicate family of YUM repositories. E.g. rhel8-BaseOS covers
+    variants of BaseOS YUM repositories for all channels, architectures,
+    RHUI, etc, which have basically same purpose.
+    """
+
+    major_version = fields.String()
+    """
+    The major version of OS.
+
+    E.g. for RHEL 7.9 the major version is 7. Regarding we work with versions
+    as with strings everywhere, we keep this data type here too.
```

Comment:
```suggestion
    E.g. for RHEL 7.9 the major version is 7. Since we work with versions
    as with strings throughout the whole codebase, we keep this data type here too.
```

Reply:
done.